mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java

```diff
@@ -0,0 +1,118 @@
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.PreDestroy;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.james.backend.rabbitmq.RabbitMQConnectionFactory;
+import org.apache.james.event.json.EventSerializer;
+import org.apache.james.mailbox.Event;
+import org.apache.james.mailbox.MailboxListener;
+
+import com.rabbitmq.client.Connection;
+
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import reactor.rabbitmq.ExchangeSpecification;
+import reactor.rabbitmq.OutboundMessage;
+import reactor.rabbitmq.RabbitFlux;
+import reactor.rabbitmq.Sender;
+import reactor.rabbitmq.SenderOptions;
+
+class RabbitMQEventBus implements EventBus {
+
+    static final String MAILBOX_EVENT = "mailboxEvent";
+    static final String MAILBOX_EVENT_EXCHANGE_NAME = MAILBOX_EVENT + "-exchange";
+    static final String EMPTY_ROUTING_KEY = "";
+
+    static final boolean DURABLE = true;
+    private static final String DIRECT_EXCHANGE = "direct";
+
+    private final EventSerializer eventSerializer;
+    private final Sender sender;
+    private final Mono<Connection> connectionMono;
+    private final Map<Group, GroupRegistration> groupRegistrations;
+
+    RabbitMQEventBus(RabbitMQConnectionFactory rabbitMQConnectionFactory, EventSerializer eventSerializer) {
+        this.connectionMono = Mono.fromSupplier(rabbitMQConnectionFactory::create).cache();
+        this.eventSerializer = eventSerializer;
+        this.sender = RabbitFlux.createSender(new SenderOptions().connectionMono(connectionMono));
+        this.groupRegistrations = new ConcurrentHashMap<>();
+    }
+
+    public Mono<Void> start() {
+        return sender.declareExchange(ExchangeSpecification.exchange(MAILBOX_EVENT_EXCHANGE_NAME)
+                .durable(DURABLE)
+                .type(DIRECT_EXCHANGE))
+            .subscribeOn(Schedulers.elastic())
+            .then();
+    }
+
+    @PreDestroy
+    public void stop() {
+        sender.close();
+        groupRegistrations.values().forEach(GroupRegistration::unregister);
+    }
+
+    @Override
+    public Registration register(MailboxListener listener, RegistrationKey key) {
+        throw new NotImplementedException("will implement latter");
+    }
+
+    @Override
+    public Registration register(MailboxListener listener, Group group) {
+        return groupRegistrations.compute(group, (groupToRegister, oldGroupRegistration) -> {
+            if (oldGroupRegistration != null) {
+                throw new GroupAlreadyRegistered(group);
+            }
+            return newRegistrationGroup(listener, groupToRegister);
+        });
+    }
+
+    private GroupRegistration newRegistrationGroup(MailboxListener listener, Group group) {
+        GroupRegistration groupRegistration = new GroupRegistration(connectionMono, sender, eventSerializer, listener, group,
+            () -> groupRegistrations.remove(group));
+        groupRegistration.registerGroup();
```

Comment:
If registerGroup returns a GroupRegistration instead of a void:
 - We would have a more fluent style
 - We could move `registerGroup` in `Registration register(MailboxListener listener, Group group)` where IMO it would make more sens (this method name only indicate the `create` not that this group is started...

```
    @Override
    public Registration register(MailboxListener listener, Group group) {
        return groupRegistrations.compute(group, (groupToRegister, oldGroupRegistration) -> {
            if (oldGroupRegistration != null) {
                throw new GroupAlreadyRegistered(group);
            }
            return newRegistrationGroup(listener, groupToRegister);
        })
        .registerGroup();
    }
```